src/common/io.py

```diff
@@ -27,3 +27,69 @@ def input_file_path(path):
     
     logging.getLogger(__name__).critical(f"Provided INPUT path {path} is neither a directory or a file???")
     return path
+
+
+class PartitioningEngine():
+    """ Class handles partitioning files into chunks with various strategies. """
```

Comment:
Maybe add here that this is line-based, i.e. groups (that need to be processed together) aren't allowed

Reply:
Note: this change has been pushed as part of another pr #58